### PR TITLE
CLN: Use more pytest idioms in test_momemts_ewm.py

### PR DESCRIPTION
**Pull request by mroeschke**
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

Use `pytest.mark.parameterize` and unnest helper test functions

### CODE REVIEW

File: pandas/tests/window/moments/test_moments_ewm.py [hunk 1]
```diff
@@ -53,63 +47,162 @@ def test_ewma_nan_handling():
...
-        if ignore_na is False:
-            # check that ignore_na defaults to False
-            result = s.ewm(com=com, adjust=adjust).mean()
-            tm.assert_series_equal(result, expected)
 
 
 @pytest.mark.parametrize("name", ["var", "vol"])
-def test_ewmvar_ewmvol(series, frame, nan_locs, name):
-    check_ew(name=name, frame=frame, series=series, nan_locs=nan_locs)
+def test_ewmvar_ewmvol(series, frame, name):
```
Comment (simonjayhawkins) [inline pandas/tests/window/moments/test_moments_ewm.py:200]: can this be combined with test_ewma

File: pandas/tests/window/moments/test_moments_ewm.py [hunk 2]
```diff
@@ -53,63 +47,162 @@ def test_ewma_nan_handling():
...
+    series_result = getattr(series.ewm(com=10), name)()
+    assert isinstance(series_result, Series)
+
+    frame_result = getattr(frame.ewm(com=10), name)()
+    assert type(frame_result) == DataFrame
```
Comment (simonjayhawkins) [inline pandas/tests/window/moments/test_moments_ewm.py:205]: why not isinstance?

File: pandas/tests/window/moments/test_moments_ewm.py [hunk 3]
```diff
@@ -53,63 +47,162 @@ def test_ewma_nan_handling():
...
-        (s0, True, False, [np.nan, (1.0 - alpha), 1.0]),
-        (s0, True, True, [np.nan, (1.0 - alpha), 1.0]),
-        (s0, False, False, [np.nan, (1.0 - alpha), alpha]),
-        (s0, False, True, [np.nan, (1.0 - alpha), alpha]),
-        (s1, True, False, [(1.0 - alpha) ** 2, np.nan, 1.0]),
-        (s1, True, True, [(1.0 - alpha), np.nan, 1.0]),
-        (s1, False, False, [(1.0 - alpha) ** 2, np.nan, alpha]),
-        (s1, False, True, [(1.0 - alpha), np.nan, alpha]),
-        (s2, True, False, [np.nan, (1.0 - alpha) ** 3, np.nan, np.nan, 1.0, np.nan]),
-        (s2, True, True, [np.nan, (1.0 - alpha), np.nan, np.nan, 1.0, np.nan]),
```
Comment (simonjayhawkins) [inline pandas/tests/window/moments/test_moments_ewm.py:77]: i'm not sure that the parameterisation is clearer.

maybe parameterise s, adjust and ignore_na independently (with fixtures) and create a `w` fixture with a dictionary lookup. i.e. pass s, adjust and ignore_na to `w` fixture

maybe use a class just for namespacing and use class level fixtures.

**Review by simonjayhawkins — COMMENTED** [review]
Thanks @mroeschke. A suggestion for an alternative approach that may not be any clearer, so feel free to ignore.

File: pandas/tests/window/moments/test_moments_ewm.py (hunk 3 above)
Comment (mroeschke) [reply pandas/tests/window/moments/test_moments_ewm.py:77]: I'll consider doing this change in a follow up PR. The main motivation with this pass is to ensure that parameterized test failures are more easily visible within pytest

**Comment by jreback** [comment]
thanks @mroeschke follows welcome (per @simonjayhawkins comments)

**Pull request merged**